File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/05.While-Loop/02-exercise/02.examPreparation.js
function examPreparation(input) {

    const lowGrades = Number(input[0]);
    let index = 1;
    let current = input[index++];

    let lowGradesCounter = 0;
    let gradeCounter = 0;
    let sumOfGrades = 0;
    let isNeedABreak = false;
    let lastProblem = "";

    while (current !== "Enough") {
        lastProblem = current;
        gradeCounter += 1;
        let currentGrade = Number(input[index++]);
        sumOfGrades += currentGrade;
        if (currentGrade <= 4) {
            lowGradesCounter += 1;
            if (lowGradesCounter === lowGrades) {
                isNeedABreak = true
                console.log(`You need a break, ${lowGradesCounter} poor grades.`);
                break;
            }
        }

        current = input[index++];
    }

    const average = sumOfGrades / gradeCounter;

    if (!isNeedABreak) {
    console.log(`Average score: ${average.toFixed(2)}`);
    console.log(`Number of problems: ${gradeCounter}`);
    console.log(`Last problem: ${lastProblem}`);
    }
}
examPreparation(["3",
"Money",
"6",
"Story",
"4",
"Spring Time",
"5",
"Bus",
"6",
"Enough"]);

File: Python_Basics/Homeworks-And-Labs/05.while_loop/02.exercise/02.exam_preparation.py
low_grades = int(input())
current = input()

low_grades_counter = 0
grade_counter = 0
sum_of_grades = 0
is_needA_break = False
last_problem = ""

while current != "Enough":
    last_problem = current
    grade_counter += 1
    currentGrade = int(input())
    sum_of_grades += currentGrade
    if currentGrade <= 4:
        low_grades_counter += 1
        if low_grades_counter == low_grades:
            is_needA_break = True
            print(f'You need a break, {low_grades_counter} poor grades.')
            break

    current = input()

average = sum_of_grades / grade_counter

if not is_needA_break:
    print(f'Average score: {average:.2f}')
    print(f'Number of problems: {grade_counter}')
    print(f'Last problem: {last_problem}')
